ccmlib/common.py

```diff
@@ -461,6 +461,37 @@ def scylla_extract_install_dir_and_mode(install_dir):
     return install_dir, scylla_mode
 
 
+def wait_for(func, timeout, first=0.0, step=1.0, text=None):
+    """
+    Wait until func() evaluates to True.
+
+    If func() evaluates to True before timeout expires, return the
+    value of func(). Otherwise return None.
```

Comment:
BTW, It's odd that `wait_for` returns `None` rather than `False`, if it's defined to wait until `func` returns `True`.

It would make sense if we defined it as `wait_until_value` where it would wait until `func` return non- `None` value (could be `False`) and returns it, otherwise return `None` if the timeout expires.

I'm not sure there's a use case for the latter though.

Reply:
It seems that in some sense, this function does return the value - the text is ambigous (and even misleading) on what "func() evaluates to True" means. It sounds like you check whether `func() == True`, but that's not actually what you do - you check `if(func())` which means that func() returned an object whose "truth value" is true (see https://docs.python.org/3/library/stdtypes.html#truth). So for example, if func() returns a string, a non-empty string is considered true, and then wait_for will return this string.

What I would have done in this case is one of two things:
1. Simplify the API. `wait_for()` will just return True or False - nothing else (not a string, not None, etc.). Does any user need more than that in practice? The API will be simpler and require less explanations in comments.
2. Keep the current return-the-value API, but fix the return value of the timeout case: In the timeout case, don't just return "None"! instead return the last "output, i.e., the last result of `func()`. This may be anything not-true - False (usually...), None, empty string or whatever. But the idea is that if the user's func() returns actual True or False, so will wait_for() instead that weird None return.

Reply:
Done

Reply:
I don't see it done, maybe you forgot to push?

Reply:
Oh,  I see, the changes are in the next patch... Please combine the two.